shell/app-shell/elements/persistent-handles.js

```diff
@@ -46,7 +46,7 @@ class PersistentHandles extends Xen.Base {
     let remoteHandleMeta = state.db.child(`views/${handleId}`);
     // TODO(sjmiles): maybe not do this unless we have to (reducing FB thrash)
     remoteHandleMeta.child('metadata').update({
-      type: ArcsUtils.metaTypeFromType(localHandle.type),
+      type: ArcsUtils.metaTypeFromType(localHandle.type.resolvedType()),
```

Comment:
Should we put this change directly in the `metaTypeFromType` impl?

Reply:
That seems like it might be better - I can't think of a case where we wouldn't want that at the ArcsUtils level?